aten/src/ATen/NamedTensorUtils.cpp

```diff
@@ -128,51 +127,41 @@ static void assert_names_equal(DimnameList a, DimnameList b) {
       ". Please rename the out tensor's dims with `Tensor.rename`.");
 }
 
-void propagate_names(TensorImpl* result, optional<DimnameList> names) {
-  if (!impl::has_names(result) && !names.has_value()) {
-    return;
-  }
-  if (!impl::has_names(result)) {
-    impl::internal_set_names_inplace(result, names);
-    return;
-  }
-  assert_names_equal(
-      impl::get_names(result),
-      names.value_or(default_names(result->dim())));
-}
-
-void propagate_names(
-    Tensor& result,
-    optional<std::vector<Dimname>>&& maybe_names,
+Tensor& propagate_names_if_nonempty(Tensor& result,
+    DimnameList maybe_names,
     bool validate_names) {
-  propagate_names(result.unsafeGetTensorImpl(), std::move(maybe_names), validate_names);
+  propagate_names_if_nonempty(result.unsafeGetTensorImpl(), maybe_names, validate_names);
+  return result;
 }
 
-void propagate_names(
-    TensorImpl* result,
-    optional<std::vector<Dimname>>&& maybe_names,
+TensorImpl* propagate_names_if_nonempty(TensorImpl* result,
+    DimnameList maybe_names,
     bool validate_names) {
-  if (!maybe_names) {
-    propagate_names(result, nullopt);
-    return;
-  }
-  propagate_names(result, std::move(maybe_names.value()), validate_names);
-}
-
-void propagate_names(TensorImpl* result, std::vector<Dimname>&& names, bool validate_names) {
-  if (!impl::has_names(result)) {
-    impl::internal_set_names_inplace(result, std::move(names), validate_names);
-    return;
+  if (maybe_names.empty()) {
+    return result;
   }
-  assert_names_equal(impl::get_names(result), names);
+  return propagate_names(result, maybe_names, validate_names);
 }
 
-void propagate_names(Tensor& result, optional<DimnameList> names) {
-  propagate_names(result.unsafeGetTensorImpl(), names);
+Tensor& propagate_names(Tensor& result, DimnameList names, bool validate_names) {
+  propagate_names(result.unsafeGetTensorImpl(), names, validate_names);
+  return result;
 }
 
-void propagate_names(Tensor& result, std::vector<Dimname>&& names, bool validate_names) {
-  propagate_names(result.unsafeGetTensorImpl(), std::move(names), validate_names);
+TensorImpl* propagate_names(TensorImpl* result, DimnameList names, bool validate_names) {
+  if (result->sizes().size() > 0) {
```

Comment:
I think `result->dim()` is clearer if I'm correctly understanding this

Reply:
Yes, thanks for catching that